### PR TITLE
update(mongodb): unified topology connection options

### PR DESCRIPTION
**Pull request by peterblazejewicz**
This commit:
- moves unified topology connection options into own interface (for
readability purposes mostly);
- adds missing properties from unified topology connection options

https://jira.mongodb.org/browse/NODE-2637
https://github.com/mongodb/node-mongodb-native/pull/2412/files
https://github.com/mongodb/node-mongodb-native/releases/tag/v3.5.9

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mongodb ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongodb@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mongodb'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. MongoTimeoutError
1. MongoServerSelectionError
1. MongoWriteConcernError
1. MongoBulkWriteError

<details>
<summary>as well as these 11 other properties...</summary>
<p>BulkWriteError, Admin, Collection, GridStore, Chunk, CoreServer, CoreConnection, Map, Symbol, BSONRegExp, instrument</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 20aef52936caf79b09107ff115057a5eaf283e6d
</p>

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -562,6 +551,57 @@ export interface DbCreateOptions extends CommonOptions {
...
+     * @default Infinity
+     */
+    maxIdleTimeMS?: number;
+    /**
+     * **Only applies to the unified topology**
+     * The maximum amount of time operation execution should wait for a connection to become available.
+     * The default is 0 which means there is no limit.
+     * @default 0
+     */
+    waitQueueTimeoutMS?: number;
```
Comment (xamgore) [inline types/mongodb/index.d.ts:602]: Could you please add break lines between properties to fit the existing code style?

**Review by xamgore — COMMENTED** [review]
Everything is fine, can be merged.

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/mongodb/index.d.ts:609]: @xamgore thx for the review! resolved with f508693d12

**Review by xamgore — APPROVED** [review]
Can be merged.

**Comment by peterblazejewicz** [comment]
Ready to merge :kiss:

**Pull request merged**